src/app/server/DefaultTermsAndConditionsProvider.cpp

```diff
@@ -0,0 +1,129 @@
+#include <platform/KeyValueStoreManager.h>
+
+namespace {
+static constexpr chip::TLV::Tag kAcceptedAcknowledgementsTag        = chip::TLV::ContextTag(1);
+static constexpr chip::TLV::Tag kAcceptedAcknowledgementsVersionTag = chip::TLV::ContextTag(2);
+static constexpr size_t kEstimatedTlvBufferSize = chip::TLV::EstimateStructOverhead(sizeof(uint16_t), sizeof(uint16_t));
+}; // namespace
+
+CHIP_ERROR chip::app::DefaultTermsAndConditionsProvider::Init(chip::PersistentStorageDelegate * const inPersistentStorageDelegate,
+                                                              uint16_t inRequiredAcknowledgementsValue,
+                                                              uint16_t inRequiredAcknowledgementsVersionValue)
+{
+    VerifyOrReturnError(nullptr != inPersistentStorageDelegate, CHIP_ERROR_INVALID_ARGUMENT);
+
+    mPersistentStorageDelegate            = inPersistentStorageDelegate;
+    mRequiredAcknowledgementsValue        = inRequiredAcknowledgementsValue;
+    mRequiredAcknowledgementsVersionValue = inRequiredAcknowledgementsVersionValue;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR chip::app::DefaultTermsAndConditionsProvider::ClearAcceptance()
+{
+    VerifyOrReturnError(nullptr != mPersistentStorageDelegate, CHIP_ERROR_UNINITIALIZED);
+
+    const chip::StorageKeyName storageKey = DefaultStorageKeyAllocator::TermsAndConditionsAcceptance();
+    ReturnErrorOnFailure(mPersistentStorageDelegate->SyncDeleteKeyValue(storageKey.KeyName()));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR chip::app::DefaultTermsAndConditionsProvider::GetAcceptance(uint16_t & outAcknowledgementsValue,
+                                                                       uint16_t & outAcknowledgementsVersionValue) const
+{
+    uint16_t acknowledgements        = 0;
+    uint16_t acknowledgementsVersion = 0;
+
+    chip::TLV::TLVReader tlvReader;
+    chip::TLV::TLVType tlvContainer;
+
+    uint8_t buffer[kEstimatedTlvBufferSize] = { 0 };
+    uint16_t bufferSize                     = sizeof(buffer);
+
+    VerifyOrReturnError(nullptr != mPersistentStorageDelegate, CHIP_ERROR_UNINITIALIZED);
+
+    const chip::StorageKeyName storageKey = DefaultStorageKeyAllocator::TermsAndConditionsAcceptance();
+    CHIP_ERROR err                        = mPersistentStorageDelegate->SyncGetKeyValue(storageKey.KeyName(), &buffer, bufferSize);
+    if (CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND == err)
+    {
+        outAcknowledgementsValue        = acknowledgements;
+        outAcknowledgementsVersionValue = acknowledgementsVersion;
+
+        return CHIP_NO_ERROR;
+    }
+
+    VerifyOrReturnError(CHIP_NO_ERROR == err, err);
+
+    tlvReader.Init(buffer);
+    ReturnErrorOnFailure(tlvReader.Next(chip::TLV::kTLVType_Structure, chip::TLV::AnonymousTag()));
+    ReturnErrorOnFailure(tlvReader.EnterContainer(tlvContainer));
+    ReturnErrorOnFailure(tlvReader.Next());
+    ReturnErrorOnFailure(tlvReader.Expect(kAcceptedAcknowledgementsTag));
```

Comment:
It might be a good idea to store a format version tag/value in here as well, in case we ever need to rev the format somehow.